R/update-input.R

```diff
@@ -642,8 +642,34 @@ updateSelectizeInput <- function(session, inputId, label = NULL, choices = NULL,
   if (!server) {
     return(updateSelectInput(session, inputId, label, choices, selected))
   }
+
+  # server side updateSelectizeInput
   value <- unname(selected)
   attr(choices, 'selected_value') <- value
+
+  # convert a single vector to a data frame so it returns {label: , value: }
+  # other objects return arbitrary JSON {x: , y: , foo: , ...}
+  choices <- if (is.atomic(choices)) {
+    # fast path
+    lab <- if(is.null(names(choices))) {
```

Comment:
The nested assignment of `lab` is somewhat confusing -- it's assigned both here on line 654, and inside the if-else, on 657. I think it would be clearer if the `lab <-` was not here, but instead on the next line, 655. Also, then line 663 could be removed.